cli/commands/service.go

```diff
@@ -0,0 +1,138 @@
+package commands
+
+import (
+	"bytes"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"encoding/json"
+
+	"fmt"
+
+	"github.com/mesosphere/dcos-commons/cli/client"
+	"github.com/mesosphere/dcos-commons/cli/config"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type DescribeHandler struct {
+	DescribeName string
+}
+
+type DescribeRequest struct {
+	AppID string `json:"appId"`
+}
+
+func unmarshallJSON(jsonBytes []byte) (map[string]interface{}, error) {
+	var responseJSON map[string]interface{}
+	err := json.Unmarshal([]byte(jsonBytes), &responseJSON)
+	if err != nil {
+		return nil, err
+	}
+	return responseJSON, nil
+}
+
+func parseDescribeResponse(responseBytes []byte) ([]byte, error) {
+	// TODO: what is the intended output here?
+	// Do we want to show upgradesTo/downgradesTo components? resolvedOptions or userProvidedOptions?
+	// TODO: is there a better way to do this instead of unmarshalling and remarshalling?
+	// TODO: add some error handling here in case the format changes
+	responseJSONBytes, err := unmarshallJSON(responseBytes)
+	if err != nil {
+		return nil, err
+	}
+	resolvedOptions, err := json.Marshal(responseJSONBytes["resolvedOptions"])
+	if err != nil {
+		return nil, err
+	}
+	return resolvedOptions, nil
+}
+
+func (cmd *DescribeHandler) DescribeConfiguration(c *kingpin.ParseContext) error {
+	// TODO: add error handling
+	requestContent, _ := json.Marshal(DescribeRequest{config.ServiceName})
+	response := client.HTTPCosmosPostJSON("describe", string(requestContent))
+	responseBytes := client.GetResponseBytes(response)
+	resolvedOptions, err := parseDescribeResponse(responseBytes)
+	if err != nil {
+		log.Printf("Failed to unmarshall response.")
+		log.Printf("Original data follows:")
+		outBuf := *bytes.NewBuffer(responseBytes)
+		outBuf.WriteTo(os.Stdout)
+		os.Exit(1)
```

Comment:
Is this our typical pattern? This seems... bad to me?

Should just return an error here and have it handled higher.